src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Network Analysis/ClosestFacility/ClosestFacility.xaml.cs

```diff
@@ -0,0 +1,171 @@
+            _incidentGraphicsOverlay = new GraphicsOverlay();
+            _facilityGraphicsOverlay = new GraphicsOverlay();
+
+            // Create a graphic and add to graphics overlay for each facility.
+            foreach (Facility facility in _facilities)
+            {
+                _facilityGraphicsOverlay.Graphics.Add(new Graphic(facility.Geometry, _facilitySymbol));
+            }
+
+            // Add each graphics overlay to MyMapView.
+            MyMapView.GraphicsOverlays.Add(_facilityGraphicsOverlay);
+            MyMapView.GraphicsOverlays.Add(_incidentGraphicsOverlay);
+
+            // Link the action of tapping on the map with the MyMapView_GeoViewTapped method.
+            MyMapView.GeoViewTapped += MyMapView_GeoViewTappedAsync;            
+
+        }
+
+        private void MyMapView_GeoViewTappedAsync(object sender, Esri.ArcGISRuntime.UI.Controls.GeoViewInputEventArgs e)
+        {
+            // Clear any prior incident and routes from the graphics.
+            _incidentGraphicsOverlay.Graphics.Clear();
+
+            // Create a MapPoint where the user clicked.
+            _incidentPoint = MyMapView.ScreenToLocation(e.Position);
+            
+            // Populate the facility parameters than solve using the task.
+            populateParametersAndSolveRouteAsync();
+
+        }
+
+        private async void populateParametersAndSolveRouteAsync()
+        {
+            // Set facilities and incident in parameters.
+            _closestFacilityParameters = _task.CreateDefaultParametersAsync().Result;
+            _closestFacilityParameters.SetFacilities(_facilities);
+            _closestFacilityParameters.SetIncidents(new List<Incident> { new Incident(_incidentPoint) } );
+
+            try
+            {
+                // Use the task to solve for the closest facility.
+                ClosestFacilityResult result = await _task.SolveClosestFacilityAsync(_closestFacilityParameters);
+
+                // Get the index of the closest facility to incident. (0) is the index of the incident, [0] is the index of the closest facility.
+                int closestFacility = result.GetRankedFacilityIndexes(0)[0];
+
+                // Get route from closest facility to the incident and display to mapview.
+                ClosestFacilityRoute route = result.GetRoute(closestFacility, 0);
+
+                // Add graphics for the incident and route.
+                _incidentGraphicsOverlay.Graphics.Add(new Graphic(route.RouteGeometry, _routeSymbol));
+                _incidentGraphicsOverlay.Graphics.Add(new Graphic(_incidentPoint, _incidentSymbol));
+
+            }
+            catch(Esri.ArcGISRuntime.Http.ArcGISWebException exception)
+            {
+                if (exception.Message.ToString().Equals("Unable to complete operation."))
+                {
+                    var message = new MessageDialog("Incident not within San Diego area!", "").ShowAsync();
```

Comment:
The team has decided not to use `var` in the public samples. While that's a bit controversial, I recommend removing it because the variable isn't needed: consider changing to `await new MessageDialog("...", "").ShowAsync();`.